pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1984,6 +1988,111 @@ private Set<String> getAllInstancesForTable(String tableNameWithType) {
     return instanceSet;
   }
 
+  /**
+   * Returns the ZK metdata for the given jobId
+   * @param jobId the id of the job
+   * @return Map representing the job's ZK properties
+   */
+  public Map<String, String> getControllerJobZKMetadata(String jobId) {
+    String controllerJobResourcePath = ZKMetadataProvider.constructPropertyStorePathForControllerJob();
+    try {
+      ZNRecord taskResourceZnRecord = _propertyStore.get(controllerJobResourcePath, null, -1);
```

Comment:
We should not use option -1 here as it is not a valid option (this is not version). We usually use `AccessOption.PERSISTENT`. Same for other places, and we don't need to explicitly handle `ZkNoNodeException`

Reply:
Ack. Yeah with AccessOption.PERSISTENT _propertyStore.get() does not throw `ZkNoNodeException`